### PR TITLE
Set has_assessments boolean for assessment template

### PR DESCRIPTION
**Pull request by k9845**
Fixes :
- #657 
- #656

### CODE REVIEW

File: apps/project/tests/test_apis.py [hunk 1]
```diff
@@ -78,6 +79,26 @@ def test_create_project(self):
...
+            ],
+            'has_assessments': True
+        }
+
+        self.authenticate()
+        response = self.client.post(url, data)
+        self.assert_201(response)
+
+        self.assertEqual(Project.objects.count(), project_count + 1)
+        self.assertEqual(response.data['assessment_template'], assessment.id)
```
Comment (thenav56) [inline apps/project/tests/test_apis.py:107]: Let's add another test for None and False as well.

File: apps/project/serializers.py [hunk 2]
```diff
@@ -377,7 +381,7 @@ class ProjectStatSerializer(ProjectSerializer):
     number_of_leads_tagged = serializers.IntegerField(read_only=True)
     number_of_leads_tagged_and_verified = serializers.IntegerField(read_only=True)
     number_of_entries = serializers.IntegerField(read_only=True)
-    status = serializers.ReadOnlyField(source='status.title')
+    status = serializers.ReadOnlyField()
```
Comment (thenav56) [inline apps/project/serializers.py]: We don't need this here. status is now a writable field.

File: apps/project/tests/test_apis.py [hunk 3]
```diff
@@ -78,6 +77,40 @@ def test_create_project(self):
...
+
+        # not providing has_assessments
+        data = {
+            'title': 'Test project',
+            'data': {'testKey': 'testValue'},
+            'organizations': [
+                {'organization': self.org1.id, 'organization_type': ProjectOrganization.DONOR},
+            ],
+            'has_assessments': False
+        }
```
Comment (thenav56) [inline apps/project/tests/test_apis.py]: This should be enough. Let's also add for None

```suggestion
        data['has_assessments'] = False
```

File: apps/project/tests/test_apis.py [hunk 4]
```diff
@@ -938,76 +971,18 @@ def test_auto_accept(self):
...
-        # One with old lead
-        project1 = self.create(Project, role=self.admin_role)
-        lead = self.create(Lead, project=project1)
-        Lead.objects.filter(pk=lead.pk).update(created_at=old_date)
-        Lead.objects.get(pk=lead.pk).save()
-
-        # One with latest lead
-        project2 = self.create(Project, role=self.admin_role)
-        self.create(Lead, project=project2)
+    def test_status_filter(self):
```
Comment (thenav56) [inline apps/project/tests/test_apis.py:982]: Let's add a test for updating status aswell.

File: apps/project/tests/test_apis.py [hunk 5]
```diff
@@ -938,76 +971,18 @@ def test_auto_accept(self):
...
-        Lead.objects.filter(pk=lead.pk).update(created_at=old_date)
-        Entry.objects.filter(pk=entry.pk).update(created_at=old_date)
-        Lead.objects.get(pk=lead.pk).save()
-
-        url = '/api/v1/projects/?status={}'.format(status.id)
+        test_user = self.create(User)
+        project1.add_member(test_user, role=self.normal_role)
 
-        self.authenticate()
+        url = f'/api/v1/projects/?status=inactive'
```
Comment (thenav56) [inline apps/project/tests/test_apis.py:990]: Let's add for active aswell.

Comment (k9845) [reply apps/project/tests/test_apis.py:990]: Done

File: apps/project/tests/test_apis.py (hunk 4 above)
Comment (k9845) [reply apps/project/tests/test_apis.py:982]: Done

File: apps/project/tests/test_apis.py (hunk 3 above)
Comment (k9845) [reply apps/project/tests/test_apis.py]: Done

File: apps/project/serializers.py (hunk 2 above)
Comment (k9845) [reply apps/project/serializers.py]: Done

File: apps/project/serializers.py [hunk 6]
```diff
@@ -224,8 +222,8 @@ class ProjectSerializer(RemoveNullFieldsMixin, DynamicFieldsMixin, UserResourceS
     user_groups = SimpleUserGroupSerializer(many=True, read_only=True)
 
     number_of_users = serializers.IntegerField(read_only=True)
-    status_title = serializers.ReadOnlyField(source='status.title')
```
Comment (thenav56) [inline apps/project/serializers.py:228]: Let's add status_display.

```python3
status_display = serializers.CharField(source='get_status_display', read_only=True)
```

Comment (k9845) [reply apps/project/serializers.py:228]: Done!

**Review by thenav56 — APPROVED** [review]

**Review by annshress — APPROVED** [review]

**Review by annshress — APPROVED** [review]
